src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -122,27 +121,35 @@ auto get_sshfs_exec_and_options(mp::SSHSession& session)
 // Split a path into existing and to-be-created parts.
 std::pair<std::string, std::string> get_path_split(mp::SSHSession& session, const std::string& target)
 {
-    QDir complete_path(QString::fromStdString(target));
-    QString absolute;
+    std::string complete_path;
 
-    if (complete_path.isRelative())
+    if ('~' == target[0])
     {
-        QString home = QString::fromStdString(run_cmd(session, "pwd")).trimmed();
-        absolute = home + '/' + complete_path.path();
+        complete_path = "~" + mp::utils::escape_all(target.substr(1, target.size() - 1));
     }
     else
     {
-        absolute = complete_path.path();
+        if ('/' != target[0])
+        {
+            complete_path = "~/" + mp::utils::escape_all(target);
+        }
+        else
+        {
+            complete_path = mp::utils::escape_all(target);
+        }
     }
 
-    QString existing =
-        QString::fromStdString(
-            run_cmd(session,
-                    fmt::format("sudo /bin/bash -c 'P=\"{}\"; while [ ! -d \"$P/\" ]; do P=${{P%/*}}; done; echo $P/'",
-                                absolute)))
-            .trimmed();
+    std::string existing =
+        run_cmd(session, fmt::format("sudo /bin/bash -c 'P={}; while [ ! -d $P/ ]; do P=${{P%/*}}; done; echo $P/'",
+                                     complete_path));
+    if ('\n' == existing.back())
+        existing.pop_back();
+
+    std::string missing = run_cmd(session, fmt::format("echo {} | cut -c {}-", complete_path, existing.size() + 1));
+    if ('\n' == missing.back())
+        missing.pop_back();
 
-    return {existing.toStdString(), QDir(existing).relativeFilePath(absolute).toStdString()};
+    return {existing, missing};
```

Comment:
Right. But note in https://doc.qt.io/qt-5/qdir.html#isRelative that 
> Under Unix a path is relative if it does not start with a "/"
so I maintain my preference for `switch(target[0])` in place of `if` and `else if`.

I normally prefer to try a `case` block instead of nested `if`'s. That said, are there any reasons I'm missing supporting your approach over mine?

Reply:
Sure, make it a `switch`.